tests/test_gabbits.py

```diff
@@ -265,6 +284,76 @@ def _generate_jwks_content(self, public_key):
         })
 
 
+class XSnippetApiWithSnippets(XSnippetApi):
+    """Start live server of XSnippet API with pre-created snippets."""
+
+    snippets = [
+        {
+            "content": "01",
+            "title": "caching decorator",
+            "syntax": "python",
+            "tags": ["decorator", "caching"],
+        },
+        {
+            "content": "02",
+            "title": "rocket content negotiation guard",
+            "syntax": "rust",
+        },
+        {
+            "content": "03",
+            "title": "auth decorator",
+            "syntax": "python",
+            "tags": ["decorator"],
+        },
+        {
+            "content": "04",
+        },
+        {
+            "content": "05",
+        },
+        {
+            "content": "06",
+        },
+        {
+            "content": "07",
+        },
+        {
+            "content": "08",
+        },
+        {
+            "content": "09",
+        },
+        {
+            "content": "10",
+        },
+    ]
+
+    def start_fixture(self):
+        super().start_fixture()
+
+        session = requests.Session()
+        endpoint = f"http://{XSNIPPET_API_HOST}:{XSNIPPET_API_PORT}/v1/snippets"
+
+        # Snippets are inserted in reverse because they are returned in
+        # descending order by default.
+        for snippet in reversed(self.snippets):
+            response = session.post(endpoint, data=json.dumps(snippet))
+            response.raise_for_status()
+
+
+class LinkHeaderResponseHandler(base.ResponseHandler):
+    """Link HTTP header response handler for Gabbi."""
+
+    test_key_suffix = "link_header"
+    test_key_value = []
+
+    def action(self, test, item, value=None):
+        item = test.replace_template(item)
+        link_items = requests.utils.parse_header_links(test.response["Link"])
+
+        test.assertIn(item, link_items)
```

Comment:
**n.b.** Unfortunately, these response handlers cannot work for strict match. :( They are evaluated for every single item in YAML independently. Hence, we will ending up parsing the link header multiple times.

Reply:
Thank you for pointing this out! I don't think we have enough test cases for this to be a major concern.

Reply:
On the other hand, there's no way to ensure that `rel` link is indeed missing (i.e. the very first page has no `prev`). I have some idea how to address that, but I'll probably do this in a separate PR.